src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
> Regarding catching errors, it's not very different 

Agreed, main difference is that by knowing a mapper was expected, we can provide a more specific error, and if we move the building of these mappings to startup time, rather than grain activation time, we can fail earlier.  Again, not much different, but slight advantage for marker attributes, IMO, slight..

> Regarding perf, it doesn't need to consider all attributes, just attributes that have a mapper associated with it.

I don't really understand what you're saying here, as the check cost I described was the lookup of the mapper in DI, which is what would be needed to consider "just attributes that have a mapper associated with it".

> I wouldn't put perf as a pro or con to keep the base attribute

Agreed, my intention was simply to prompt an examination of the potential differences, especially in services that use other containers which may provide their own attribution of parameters (I'd not suspect as being terribly common).  I don't see the perf differences as significant either.

Let's see if we can agree on what the Pros/Cons are:
Using marker attribute.
Pros:

- Allows more specific error handling.
- Allows for checking at silo start.
- Can't repurpose preexisting or third party attributes.

Cons:

- Attributes require dependency on Orleans.
- Can't repurpose preexisting or third party attributes.